### ISSUE TITLE
Installing NodeBB as a Windows Service

### ISSUE THREAD

**Issue by EnDylan** [NodeBB/NodeBB#5807]
Feature Request
---
A feature able to install NodeBB as a Windows Service so that NodeBB will run in the **background** and automatically start once the computer has booted up.

Concept
---
- Type `nodebb service -install` to install NodeBB service

Once installed, `nodebb start` and all `nodebb` commands are global and the NodeBB instance be started with a command prompt in any directory.

**Comment by pitaj** [comment]
I can see the service install thing happening, but I don't see the "commands are global" thing happening, unless it's just a part of being a Windows service (I'm ignorant on how exactly services work on Windows). 

It could be possible to just write a Powershell script, and execute that on Windows when `nodebb service install` is run. If it is trivial to do this in the first place, it may be more of a documentation thing, since that's how we handle it with Linux. 

I don't _really_ like the idea of setting up services automatically, especially since it requires admin privileges.

**Comment by EnDylan** [comment]
Okay. Forget about the global. Now that I noticed, it may cause problems if there's two instances.

I agree about the Powershell script part. Maybe a command that just runs nodebb in the background. 
So it is just possible to start the script on boot with nodebb running without a cmd prompt.

Basically. Just running NodeBB in the background because on Windows there will be an annoying cmd for NodeBB.

**Comment by Radiergummi** [comment]
Have you looked at https://github.com/coreybutler/node-windows? I'm using it to run several node applications in enterprise environments on Windows Server 2008 and 2012 R2. Running smoothly since about one and a half year :)

**Comment by EnDylan** [comment]
I think that NodeBB should have this built it. Much easier than searching around. It will be easier for noobs. Or maybe. Just start and run it in the background instead of having a redundant cmd prompt that does absolutely nothing other than running NodeBB

**Comment by pitaj** [comment]
The `./nodebb` utility does much more than just managing NodeBB processes.

**Comment by EnDylan** [comment]
Yes. But on windows, when you do `./nodebb start`, a redundant cmd will appear, just to run NodeBB. Why not in the background? and don't open an extra cmd just for that.

**Comment by Radiergummi** [comment]
Running web services in Windows is a bad idea unless you are an IIS professional. There is no reason to build a complex service abstraction into the NodeBB core. If you want it running as a daemon, either use node-service or leave the CMD prompt open (shouldn't be a problem on a server anyway).

**Comment by pitaj** [comment]
I believe that's due to how the daemon spawner works. I use Powershell, and don't see that happening when I run `./nodebb start`

**Comment by Radiergummi** [comment]
@pitaj changing NodeBB.bat to `start /b "node ./NodeBB"` could potentially fix the additional window, though I'd really rather recommend node-windows

**Comment by julianlam** [comment]
While in general we want NodeBB to be installable everywhere, it is **best** installed on a headless linux server, meaning Windows does often get the second-class treatment.

If `./nodebb start` spawns a separate daemon window, that's Windows behaviour and not something we can stop, and so if the best workaround is to install NodeBB as a service, then it should be done **manually** by the user themselves, and not automatically (as pointed out earlier, this may conflict in cases where multiple NodeBBs are running on the same system).

**Comment by julianlam** [comment]
I'd recommend changes to the documentation be made here: https://docs.nodebb.org/installing/os/windows8/

If somebody wants to write up the blurb here with instructions on how to add NodeBB as a windows service, I'll add it to the documentation.

**Comment by nodebb-misty** [comment]
Issue closed due to inactivity.

----

*This is an automated message. If you feel this action was in error, please comment on this issue so it can be looked at again*

**Comment by a632079** [comment]
I recommend to use a package manager to install dependences.(e.g chocolatey)
You can install these packages by a commond:`cinst -y packages`.
I have submit a pullrequest few months ago.But my English is not well,might many sentences is hardly to understand.
